### PR TITLE
Initialize default PVP parameters

### PR DESCRIPTION
**Pull request by vkrishna1997**

### CODE REVIEW

File: six/modules/c++/cphd/include/cphd/PVP.h [hunk 1]
```diff
@@ -396,22 +415,36 @@ struct Pvp
...
-     *  \brief Validate and set the metadata of the specified parameter
+     *  \param hasAmpSF flag indicates if optional param AmpSF exists
+     *  \param hasFxN1 flag indicates if optional param FxN1 exists
+     *  \param hasFxN2 flag indicates if optional param FxN2 exists
+     *  \param hasToaE1 flag indicates if optional param ToaE1 exists
+     *  \param hasToaE2 flag indicates if optional param ToaE2 exists
+     *  \param hasTDIonoSRP flag indicates if optional param TDIonoSRP exists
+     *  \param hasSignal flag indicates if optional param signal exists
+     */
+    void specifyOptionalParameters(bool hasAmpSF = false, bool hasFxN1 = false, bool hasFxN2 = false,
```
Comment (JonathanMeans) [inline six/modules/c++/cphd/include/cphd/PVP.h]: Now that I actually see it, I think having a function that takes 7 boolean arguments is going to be hard to read and easy to misuse. Can you find a more explicit interface, a separate function for each parameter or passing flags, or something?

(And yeah, I realize this was my bad idea...)

File: six/modules/c++/cphd/source/PVP.cpp [hunk 2]
```diff
@@ -42,8 +43,25 @@ APVPType::APVPType() :
...
+    initialize();
+    specifyOptionalParameters(hasAmpSF, hasFxN1, hasFxN2,
+                              hasToaE1, hasToaE2, hasTDIonoSRP,
+                              hasSignal);
+}
+
+void Pvp::initialize()
+{
+    // Rest of the parameters are already initialized
+    // in PVPType constructor
```
Comment (JonathanMeans) [inline six/modules/c++/cphd/source/PVP.cpp]: I'd rather have everything specified here, so the information is all in one place in the code.

File: six/modules/c++/cphd/include/cphd/PVP.h [hunk 3]
```diff
@@ -435,6 +501,27 @@ struct Pvp
...
+
+    /*
+     * Set default size and format for each parameter
+     */
+    void setDefaultValues(size_t size, const std::string& format, PVPType& param);
+
+    /*
+     *  Optional parameter flags
+     */
+    bool mAmpSF;
```
Comment (JonathanMeans) [inline six/modules/c++/cphd/include/cphd/PVP.h]: Can you make it work without this? I'd like checking for optional parameters to be consistent across the library, if possible.

Comment (vkrishna1997) [reply six/modules/c++/cphd/include/cphd/PVP.h]: CPHD03::VBM.h does this too so I was trying to be consistent with that, although CPHD03 does only has 3 optional parameters. I can move the flags to PvpBlock which is the CHPD10 equivalent of VBM.h if that's slightly better.
Or did you have some other case of optional parameter handling elsewhere in mind?

Comment (JonathanMeans) [reply six/modules/c++/cphd/include/cphd/PVP.h]: Fair, that makes sense.

File: six/modules/c++/cphd/source/PVP.cpp [hunk 4]
```diff
@@ -31,6 +31,7 @@ namespace cphd
 const size_t PVPType::WORD_BYTE_SIZE = 8;
 
 PVPType::PVPType() :
+    // Overwrite these default values if required
```
Comment (JonathanMeans) [inline six/modules/c++/cphd/source/PVP.cpp]: This comment is stale, right/

**Pull request merged**